LDM-592.tex

```diff
@@ -0,0 +1,1202 @@
+}
+
+\usecase{DRP16}{Pre-cache data}{%
+
+\item
+A Staff Scientist or Astronomer is going to a conference and wants to pre-cache some data, from a remote repository, to be able to access through the butler while without network connectivity.
+}
+
+\usecase{DRP17}{Shared cache}{%
+
+\item
+A Staff Scientist or Astronomer and her colleagues want to access the same data (or overlapping) from a remote repository. It would be efficient if this can be cached in an on-site proxy.
+}
+
+\usecase{DRP18}{Load size}{%
+
+\item
+A Staff Scientist or Astronomer (or the task they are running) wants to be able to access a dataset that is too large to fit in memory as a whole.
+}
+
+\usecase{DRP22}{Transactions}{%
+
+\item
+A SuperTask or the Batch Control System needs to write a dataset to a repository which is backed by multiple media (e.g., part goes to file, part to a database).
+This needs to happen atomically in a single blocking ACID transaction.
+Either all of the data is written, or none of it is (and an exception is thrown).
+i.e., it should offer the strong exception guarantee.
+}
+
+\usecase{DRP23}{User interactions}{%
+
+\item
+An Astronomer wants to build a query interactively (e.g., using tab-complete or some other predictive help that knows about the dataflow).
+}
+
+\usecase{DRP26}{Compression}{%
+
+\item
+When running a Pipeline in a certain environment, a developer or operator wants to configure a particular output dataset to be written with compression.
+This may be implemented by adding a new dataset type and (trivial) SuperTask to write the compressed version (as long as some sort of alias feature allows consuming code to read either the compressed or uncompressed version), or by changing Butler configuration.
+}
+
+\usecase{DRP27}{Iterating over Repository content}{%
+
+\item
+An Astronomer wants to loop over all Datasets of a particular DatasetType in a repository (e.g. to run jointcal and then make plots for all output).
+}
+
+\usecase{DRP28}{Backwards compatibility}{%
+
+\item
+A Staff Scientist or Astronomer has existing code (written to use \texttt{butler.get()} and \texttt{butler.put()} calls) that they wish to use with no (or minimal) changes.
+}
+
+\usecase{DRP29}{Multi-Repository File Removal}{%
+
+\item
+A shared storage space used for collaborative algorithm development and testing is filling up, and developers respond by removing processing outputs that are not being used anymore.
+Some of the processing runs to be deleted were based on inputs produced in other processing runs that should not be deleted, and those inputs are hence considered to be part of both the to-be-deleted data repositories and the to-be-kept data repositories.
+After removing the to-be-deleted data repositories, the files that are no longer used by any repository are identified and deleted, while those in the to-be-kept repositories are not.
+}
```

Comment:
Text for a new use-case:
```
\usecase{DRP30}{Forced deletion of datasets}

\item
One or more data products are deleted by a production operator or other authorized actor, without regard for which data repositories or other users may refer to them.  In some contexts this may not remove all record of these datasets held outside their primary storage locations, as long as the storage cost of those records is negligible.
```